File: frontend/src/api/briefing.ts
import { api, AI_TIMEOUT } from './client';

export interface BriefingTask {
  id: number;
  title: string;
  due_date: string | null;
  priority: string;
  course_name: string | null;
  is_overdue: boolean;
}

export interface BriefingAssignment {
  id: number;
  title: string;
  due_date: string | null;
  course_name: string;
  max_points: number | null;
  status: string;
  is_late: boolean;
}

export interface BriefingChildSection {
  student_id: number;
  full_name: string;
  grade_level: number | null;
  overdue_tasks: BriefingTask[];
  due_today_tasks: BriefingTask[];
  upcoming_assignments: BriefingAssignment[];
  recent_study_count: number;
  needs_attention: boolean;
}

export interface DailyBriefingResponse {
  date: string;
  greeting: string;
  children: BriefingChildSection[];
  total_overdue: number;
  total_due_today: number;
  total_upcoming: number;
  attention_needed: boolean;
}

export interface HelpMyKidRequest {
  student_id: number;
  item_type: 'task' | 'assignment';
  item_id: number;
}

export interface HelpMyKidResponse {
  study_guide_id: number;
  title: string;
}

export const briefingApi = {
  getDaily: async () => {
    const response = await api.get('/api/briefing/daily');
    return response.data as DailyBriefingResponse;
  },
  helpMyKid: async (data: HelpMyKidRequest) => {
    const response = await api.post('/api/briefing/help-my-kid', data, AI_TIMEOUT);
    return response.data as HelpMyKidResponse;
  },
};

File: app/schemas/briefing.py
"""Schemas for the Daily Briefing endpoint."""

from datetime import datetime
from pydantic import BaseModel


class BriefingTask(BaseModel):
    id: int
    title: str
    due_date: datetime | None = None
    priority: str = "medium"
    course_name: str | None = None
    is_overdue: bool = False


class BriefingAssignment(BaseModel):
    id: int
    title: str
    due_date: datetime | None = None
    course_name: str
    max_points: float | None = None
    status: str = "pending"  # pending, submitted, graded
    is_late: bool = False


class ChildBriefing(BaseModel):
    student_id: int
    full_name: str
    grade_level: int | None = None
    overdue_tasks: list[BriefingTask] = []
    due_today_tasks: list[BriefingTask] = []
    upcoming_assignments: list[BriefingAssignment] = []
    recent_study_count: int = 0  # study guides created in last 7 days
    needs_attention: bool = False  # True if overdue items exist


class DailyBriefingResponse(BaseModel):
    """Single-call response for the parent Daily Briefing card."""
    date: str  # ISO date string
    greeting: str  # e.g. "Good morning, Sarah"
    children: list[ChildBriefing] = []
    total_overdue: int = 0
    total_due_today: int = 0
    total_upcoming: int = 0
    attention_needed: bool = False  # True if any child needs attention


class HelpMyKidRequest(BaseModel):
    """Request to generate a study guide for a parent's child."""
    student_id: int
    item_type: str  # "task" or "assignment"
    item_id: int


class HelpMyKidResponse(BaseModel):
    """Response after generating a study guide via Help My Kid."""
    study_guide_id: int
    title: str
    safety_checked: bool = True
